src/HttpClient/HttpClient.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\HttpClient;
+
+use Sentry\Options;
+use Sentry\Util\Http;
+
+/**
+ * @internal
+ */
+class HttpClient implements HttpClientInterface
+{
+    /**
+     * @var string The Sentry SDK identifier
+     */
+    protected $sdkIdentifier;
+
+    /**
+     * @var string The Sentry SDK identifier
+     */
+    protected $sdkVersion;
+
+    public function __construct(string $sdkIdentifier, string $sdkVersion)
+    {
+        $this->sdkIdentifier = $sdkIdentifier;
+        $this->sdkVersion = $sdkVersion;
+    }
+
+    public function sendRequest(string $requestData, Options $options): Response
+    {
+        $dsn = $options->getDsn();
+        if (null === $dsn) {
+            throw new \RuntimeException('The DSN option must be set to use the HttpClient.');
+        }
+
+        $curlHandle = curl_init();
+        curl_setopt($curlHandle, \CURLOPT_URL, $dsn->getEnvelopeApiEndpointUrl());
+        curl_setopt($curlHandle, \CURLOPT_HTTPHEADER, Http::getRequestHeaders($dsn, $this->sdkIdentifier, $this->sdkVersion));
+        curl_setopt($curlHandle, \CURLOPT_USERAGENT, $this->sdkIdentifier . '/' . $this->sdkVersion);
+        curl_setopt($curlHandle, \CURLOPT_TIMEOUT, $options->getHttpTimeout());
+        curl_setopt($curlHandle, \CURLOPT_CONNECTTIMEOUT, $options->getHttpConnectTimeout());
+        curl_setopt($curlHandle, \CURLOPT_ENCODING, '');
+        curl_setopt($curlHandle, \CURLOPT_POST, true);
+        curl_setopt($curlHandle, \CURLOPT_POSTFIELDS, $requestData);
+        curl_setopt($curlHandle, \CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($curlHandle, \CURLOPT_HEADER, true);
+        curl_setopt($curlHandle, \CURLOPT_HTTP_VERSION, \CURL_HTTP_VERSION_1_1);
+
+        $httpSslVerifyPeer = $options->getHttpSslVerifyPeer();
+        if ($httpSslVerifyPeer) {
+            curl_setopt($curlHandle, \CURLOPT_SSL_VERIFYPEER, true);
+        }
+
+        $httpProxy = $options->getHttpProxy();
+        if (null !== $httpProxy) {
+            curl_setopt($curlHandle, \CURLOPT_PROXY, $httpProxy);
```

Comment:
We might also want to set `CURLOPT_HEADEROPT` to `CURLHEADER_SEPARATE` to be safe. Before cURL `7.42.1` `CURLOPT_HTTPHEADER` was also sent to the proxy which might cause issues, in later versions `CURLHEADER_SEPARATE` became the standard but since we don't know the cURL version we might want to set this ourselfs.